pkg/provisioning/bootguard/bootguard.go

```diff
@@ -888,20 +888,41 @@ func (b *BootGuard) BPMKeyMatchKMHash() (bool, error) {
 	return true, nil
 }
 
-// SaneBPMSecurityProps verifies that BPM contains security properties set accordingly to spec
-func (b *BootGuard) SaneBPMSecurityProps() (bool, error) {
+func (b *BootGuard) StrictSaneBPMSecurityProps() (bool, error) {
 	switch b.Version {
 	case bgheader.Version10:
 		flags := b.VData.BGbpm.SE[0].Flags
-		if !flags.DMAProtection() {
-			return false, fmt.Errorf("dma protection should be enabled for bootguard")
-		}
 		if !flags.AuthorityMeasure() {
 			return false, fmt.Errorf("pcr-7 data should extended for OS security")
 		}
 		if !flags.TPMFailureLeavesHierarchiesEnabled() {
 			return false, fmt.Errorf("tpm failure should lead to default measurements from PCR0 to PCR7")
 		}
+	case bgheader.Version20:
+		bgFlags := b.VData.CBNTbpm.SE[0].Flags
+		if !bgFlags.AuthorityMeasure() {
+			return false, fmt.Errorf("pcr-7 data should extended for OS security")
+		}
+		if !bgFlags.TPMFailureLeavesHierarchiesEnabled() {
+			return false, fmt.Errorf("tpm failure should lead to default measurements from PCR0 to PCR7")
+		}
+		txtFlags := b.VData.CBNTbpm.TXTE.ControlFlags
+		if txtFlags.MemoryScrubbingPolicy() != cbntbootpolicy.MemoryScrubbingPolicySACM {
+			return false, fmt.Errorf("S-ACM memory scrubbing should be used over the BIOS")
+		}
+	}
+
+	return b.SaneBPMSecurityProps()
+}
+
+// SaneBPMSecurityProps verifies that BPM contains security properties set accordingly to spec
+func (b *BootGuard) SaneBPMSecurityProps() (bool, error) {
+	switch b.Version {
+	case bgheader.Version10:
+		flags := b.VData.BGbpm.SE[0].Flags
+		if !flags.DMAProtection() {
+			return false, fmt.Errorf("dma protection should be enabled for bootguard")
+		}
```

Comment:
If this is not strict: Why is this function here then?

Reply:
See above.